replica/table.cc

```diff
@@ -716,23 +724,28 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
                 _memtables->erase(old);
                 tlogger.debug("Memtable for {}.{} replaced, into {} sstables", old->schema()->ks_name(), old->schema()->cf_name(), newtabs.size());
                 co_return stop_iteration::yes;
-            } catch (const std::exception& e) {
-                for (auto& newtab : newtabs) {
-                    newtab->mark_for_deletion();
-                    tlogger.error("failed to write sstable {}: {}", newtab->get_filename(), e);
-                }
-                _config.cf_stats->failed_memtables_flushes_count++;
-                // If we failed this write we will try the write again and that will create a new flush reader
-                // that will decrease dirty memory again. So we need to reset the accounting.
-                old->revert_flushed_memory();
-                co_return stop_iteration(_async_gate.is_closed());
+            } catch (...) {
+                ex = std::current_exception();
             }
+            co_return co_await handle_error(ex, std::move(newtabs), std::move(old));
         };
         co_return co_await with_scheduling_group(default_scheduling_group(), std::ref(post_flush));
     };
     co_return co_await with_scheduling_group(_config.memtable_scheduling_group, std::ref(try_flush));
 }
 
+future<stop_iteration> table::handle_error(const std::exception_ptr& e, std::vector<sstables::shared_sstable> newtabs, lw_shared_ptr<memtable> old) {
```

Comment:
looks like we can inline the exception handling using std::rethrow_exception. this function won't be used anywhere else, so I think it's better to not have this function, but inline it instead.